File: Swift/프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.swift
import Foundation

func solution(_ l:Int, _ r:Int) -> [Int] {
    var result = [Int]()
    
     func generateNumbers(_ current: String) {
        if let number = Int(current), number >= l, number <= r {
            result.append(number)
        }
        
        if let number = Int(current), number > r {
            return
        }

        generateNumbers(current + "0")
        generateNumbers(current + "5")
    }
    
    generateNumbers("5")
    
    result.sort()
    
    return result.isEmpty ? [-1] : result
}

File: Python3/프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    answer = []
    
    def generateNumber(current):
        number = int(current)
        
        if l <= number <= r:
            answer.append(number)
        if number > r:
            return
        
        generateNumber(current + "0")
        generateNumber(current + "5")
    
    generateNumber("5")
    
    answer.sort()
            
    return answer if answer else [-1]